### PR TITLE
Add test data for example workflow notebooks

### PR DESCRIPTION
**Pull request by geek-yang**
In this PR we add test data for example workflow notebooks. The data is originally from ERA5. Since downloading and preprocessing the data could take time, we provide the processed data in this repo to make it easier for users to play with the notebooks.

We also add a README.md about how the data is generated, with a notebook containing all steps for users who want to re-produce the data themselves.

### CODE REVIEW

**Comment by geek-yang** [comment]
To review this PR, please check README.md and try to run notebook `prepare_test_data.ipynb`.

**Review by geek-yang — COMMENTED** [review]
To review the workflow notebooks, just pick up one of them and try to see whether it runs on your machine or not is fine. I only replace the data and have tested all of them on my machine.

File: data/prepare_test_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,3407 @@
...
+       "    source:   ERA5"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "cluster_mask = xr.open_dataset('t2m_daily_1950-2021_2deg_clustered_226_300E_30_70N.nc')\n",
```
Comment (BSchilperoort) [inline data/prepare_test_data.ipynb:1493]:
```suggestion
    "cluster_mask = xr.open_dataset('t2m_daily_1959-2021_2deg_clustered_226_300E_30_70N.nc')\n",
```
The file included here has "1959" as the start year, not 1950.

File: data/prepare_test_data.ipynb [hunk 2]
```diff
@@ -0,0 +1,3407 @@
...
+    "Export processed data as netcdf4 file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t2m_clusters_dataset.to_netcdf(\"t2m_daily_1950-2021_2deg_clustered_226_300E_30_70N_copy.nc\")"
```
Comment (BSchilperoort) [inline data/prepare_test_data.ipynb:3381]:
```suggestion
    "t2m_clusters_dataset.to_netcdf(\"t2m_daily_1959-2021_2deg_clustered_226_300E_30_70N_copy.nc\")"
```

**Review by BSchilperoort — APPROVED** [review]
I only spotted some small mistakes with the prepare_test_data notebook.

Also, looking at the LSTM results:
image
It seems like the model is just predicting the mean... :thinking: 
It would be good to compare the model results to just the expected mean.

Otherwise, things seem to run. I only ran the ridge notebook, as the others require wandb.

**Comment by geek-yang** [comment]
> I only spotted some small mistakes with the prepare_test_data notebook.
> 
> Also, looking at the LSTM results: image It seems like the model is just predicting the mean... 🤔 It would be good to compare the model results to just the expected mean.
> 
> Otherwise, things seem to run. I only ran the ridge notebook, as the others require wandb.

Thanks for the notice. Indeed I planned to further inspect it in the comparison notebook. But I agree that it seems LSTM just predicts the mean. I further tune the hyper-parameters a bit and it seems the model is under-trained. After taking more epochs for training, the results got improved. See the figure below (also including the expected mean for now). I will try to further improve it in the comparison one (https://github.com/AI4S2S/cookbook/pull/8).

image

**Pull request merged**